### DISCUSSION TITLE
国产arm平台源码安装求助

### DISCUSSION THREAD

**Discussion by jianganhong** [milvus-io/milvus#4859] [Q&A and General discussion]
可以帮助我在国产arm平台飞腾CPU的kylin系统（基于centos8）的milvus源码安装吗

**Comment by yhmo** [comment]
我们之前尝试过在某些arm平台上编译milvus，非常困难，编译错误千奇百怪，最后我们放弃了。主要是mysql的客户端和faiss在arm上比较难编译。你可以在issue里搜索关键字 “arm”，有几条已经关闭的issue都是关于arm上编译milvus的。

**Reply by jianganhong** [reply]
您好。官方另一种办法docker下安装您有试过吗。

**Reply by yhmo** [reply]
我门没有在ARM平台上试过。如果你们的ARM系统上可以运行docker的话，可以一试。

**Reply by yhmo** [reply]
这里是在docker里编译milvus gpu版本的文章：https://zhuanlan.zhihu.com/p/272746341
你们可能编译cpu版本就可以。不需要带cuda的镜像。
编译的时候可能需要较多的内存，所以机器的内存最好能高点。用centos x86的系统编译是没问题的，我们的官方镜像用的就是centors的系统。

**Reply by jianganhong** [reply]
你好我安装了docker19.03版本 然后安装milvus官网安装方法安装后运行报错，可以帮我看一下吗？
[root@node01 conf]# docker image
Usage:  docker image COMMAND
Manage images
Commands:
  build       Build an image from a Dockerfile
  history     Show the history of an image
  import      Import the contents from a tarball to create a filesystem image
  inspect     Display detailed information on one or more images
  load        Load an image from a tar archive or STDIN
  ls          List images
  prune       Remove unused images
  pull        Pull an image or a repository from a registry
  push        Push an image or a repository to a registry
  rm          Remove one or more images
  save        Save one or more images to a tar archive (streamed to STDOUT by default)
  tag         Create a tag TARGET_IMAGE that refers to SOURCE_IMAGE

Run 'docker image COMMAND --help' for more information on a command.

[root@node01 conf]# docker images
REPOSITORY          TAG                        IMAGE ID            CREATED             SIZE
milvusdb/milvus     1.0.0-cpu-d030521-1ea92e   de52c8960058        3 weeks ago         714MB

[root@node01 conf]# ls
server_config.yaml

[root@node01 conf]# cd ..

[root@node01 milvus]# ls
conf  db  logs  wal

[root@node01 milvus]# pwd
/home/root/milvus

[root@node01 milvus]# cd /home/root/milvus/
[root@node01 milvus]# ls
conf  db  logs  wal
[root@node01 milvus]# ls db/
[root@node01 milvus]# ls conf/
server_config.yaml
[root@node01 milvus]# ls logs/
[root@node01 milvus]# ls wal/
[root@node01 milvus]# du -sh ..
20K     ..

[root@node01 milvus]#  docker logs milvus_cpu_1.0.0
standard_init_linux.go:211: exec user process caused "exec format error"
[root@node01 milvus]#



[root@node01 milvus]# docker info
Client:
 Debug Mode: false
Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 1
 Server Version: 19.03.0
 Storage Driver: overlay2
  Backing Filesystem: xfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 894b81a4b802e4eb2a91d1ce216b8817763c29fb
 runc version: 425e105d5a03fabd737a126ad93d62a9eeede87f
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.90-17.ky10.aarch64
 Operating System: Kylin Linux Advanced Server V10 (Tercel)
 OSType: linux
 Architecture: aarch64
 CPUs: 64
 Total Memory: 125.7GiB
 Name: node01
 ID: D7Q5:YKMF:NNKU:QFYZ:VYV7:NEBR:W2R5:HFXP:EDUW:FEHP:3JHG:D3AG
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine

**Reply by yhmo** [reply]
我们这个镜像是在x86下build出来的，没法跑在arm平台下。你得尝试在arm平台下编译milvus。

**Reply by jianganhong** [reply]
明白啦，再试试

**Reply by jianganhong** [reply]
明白啦，再试试

**Reply by jianganhong** [reply]
你好。在x86 ubuntu源码编译
[ 37%] Completed 'prometheus_ep'
[ 37%] Built target prometheus_ep
make: *** [Makefile:130：all] 错误 2
这是什么原因呀

**Reply by JackLCL** [reply]
@jianganhong 为了方便大家更及时的沟通，欢迎加入我们的milvus技术交流群~
![image](https://user-images.githubusercontent.com/53512883/112792351-ddc2f380-9095-11eb-8054-46cbccd7c3fd.png)

**Reply by yhmo** [reply]
> 你好。在x86 ubuntu源码编译
> [ 37%] Completed 'prometheus_ep'
> [ 37%] Built target prometheus_ep
> make: *** [Makefile:130：all] 错误 2
> 这是什么原因呀

编译错误没贴全，仅从这几行看不出原因

**Comment by caozibiao** [comment]
安装milvus集群时，etcd和minio报错： exec user process caused "exec format error"

**Reply by xiaofan-luan** [reply]
> 安装milvus集群时，etcd和minio报错： exec user process caused "exec format error"

目前社区版本不会提供arm编译的支持，不过我们成功跑起来过。如果有企业版的需求欢迎联系我们

**Reply by GreateFang** [reply]
请问问题解决了吗
